sky/spot/controller.py

```diff
@@ -177,13 +182,26 @@ def run(self):
         except exceptions.ResourcesUnavailableError as e:
             logger.error(f'{common_utils.class_fullname(e.__class__)}: '
                          f'{colorama.Fore.RED}{e}{colorama.Style.RESET_ALL}')
-            spot_state.set_failed(
-                self._job_id,
-                failure_type=spot_state.SpotStatus.FAILED_NO_RESOURCE)
+            failure_type = spot_state.SpotStatus.FAILED_NO_RESOURCE
+            failure_reason = str(e)
+            if len(e.failover_reasons) == 1:
```

Comment:
Discussed: probably want comments or rewrite this expression

Reply:
Refactored this to check whether there is `ResourcesUnavailbableError` in the `failover_reasons`. PTAL. : )